test/unit/disposable_common.dart

```diff
@@ -60,6 +60,29 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
       });
       disposable.dispose();
     });
+
+    test('should throw during disposal', () {
+      var completer = new Completer();
+      disposable.awaitBeforeDispose(completer.future);
+      disposable.dispose();
+      new Future(() {}).then((_) {
```

Comment:
need to await this or use `expectAsync()` otherwise the `then()` callback can/will execute after the test is finished